sign.go

```diff
@@ -0,0 +1,82 @@
+package floodsub
+
+import (
+	"fmt"
+
+	pb "github.com/libp2p/go-floodsub/pb"
+
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+func verifyMessageSignature(m *pb.Message) error {
+	var pubk crypto.PubKey
+
+	pid, err := peer.IDFromBytes(m.From)
+	if err != nil {
+		return err
+	}
+
+	if m.Key == nil {
```

Comment:
i'd pull this whole logic out into a 'get key' method either on the message, or that takes a message

Reply:
ok

Reply:
done